### PR TITLE
[LINKIS #1632] Add linkis-engineplugin-elasticsearch module

### LINKED ISSUE #1632: [Feature] Enhance ElasticSearchEngine, adatps to Linkis1.X architecture
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-linkis/issues) and found no similar feature requirement.

### Problem Description

Follow up to #711.

### Description

ElasticSearchEngine adapts to the Linkis1.X's architecture, Implement the ElasticSearchEngineConn.

### Use case

_No response_

### solutions

subtasks:
+ [ ] Merge the master branch, rename the package, and do some adaptations.
+ [ ] Add tests for ElasticSearchEngine.
+ [ ] Add documentation for ElasticSearchEngine.

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by wForget**
### What is the purpose of the change

Enhance ElasticSearchEngine, adatps to Linkis1.X architecture.
close #1632 

### Brief change log
- Implement the `linkis-engineplugin-elasticsearch` module according to the engine development specification.

### Verifying this change
- Using `UJESClient`, submit a elasticsearch json query task to verify it.

### Does this pull request potentially affect one of the following parts:
- Dependencies (does it add or upgrade a dependency): (yes)
- Anything that affects deployment: (no)
- The MGS(Microservice Governance Services), i.e., Spring Cloud Gateway, OpenFeign, Eureka.: (no)

### Documentation
- Does this pull request introduce a new feature? (yes)
- If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: linkis-engineconn-plugins/engineconn-plugins/elasticsearch/pom.xml [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        <version>1.1.2</version>
+        <relativePath>../../../pom.xml</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>linkis-engineplugin-elasticsearch</artifactId>
+
+    <properties>
+        <elasticsearch.version>7.6.2</elasticsearch.version>
+        <fasterxml.jackson.version>2.11.3</fasterxml.jackson.version>
```
Comment (peacewong) [inline linkis-engineconn-plugins/engineconn-plugins/elasticsearch/pom.xml:34]: It is recommended to use the jackson version in the main pom

File: linkis-engineconn-plugins/engineconn-plugins/elasticsearch/src/main/scala/org/apache/linkis/engineplugin/elasticsearch/executer/client/impl/ResponseHandlerImpl.scala [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    // write result
+    if (isTable) {
+      ElasticSearchTableResponse(columns.toArray, records.toArray)
+    } else {
+      ElasticSearchJsonResponse(new String(contentBytes, charSet))
+    }
+  }
+  // scalastyle:on
+
+//  def writeText(content: String, storePath: String, alias: String, proxyUser: String): String = {
```
Comment (peacewong) [inline linkis-engineconn-plugins/engineconn-plugins/elasticsearch/src/main/scala/org/apache/linkis/engineplugin/elasticsearch/executer/client/impl/ResponseHandlerImpl.scala:158]: It is recommended to remove the commented code

File: linkis-engineconn-plugins/engineconn-plugins/elasticsearch/src/main/scala/org/apache/linkis/engineplugin/elasticsearch/conf/ElasticSearchConfiguration.scala [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ */
+package org.apache.linkis.engineplugin.elasticsearch.conf
+
+import org.apache.linkis.common.conf.{ByteType, CommonVars}
+
+object ElasticSearchConfiguration {
+
+  // es client
+  val ES_CLUSTER = CommonVars("wds.linkis.es.cluster", "127.0.0.1:9200")
+  val ES_DATASOURCE_NAME = CommonVars("wds.linkis.datasource", "default_datasource")
```
Comment (casionone) [inline linkis-engineconn-plugins/engineconn-plugins/elasticsearch/src/main/scala/org/apache/linkis/engineplugin/elasticsearch/conf/ElasticSearchConfiguration.scala:25]: The new configuration item name is recommended not to be prefixed with wds.

Comment (wForget) [reply linkis-engineconn-plugins/engineconn-plugins/elasticsearch/src/main/scala/org/apache/linkis/engineplugin/elasticsearch/conf/ElasticSearchConfiguration.scala]: thanks, fixed

**Pull request merged**